pkgs/development/compilers/swift/swift-driver/default.nix

```diff
@@ -0,0 +1,81 @@
+{ lib
+, stdenv
+, callPackage
+, fetchFromGitHub
+, fetchpatch
+, swift
+, swiftpm
+, Foundation
+, XCTest
+, sqlite
+, ncurses
+, substituteAll
+}:
+let
+  # Generated by swiftpm2nix.
+  generated = callPackage ./generated { };
+
+  # On Darwin, we only want ncurses in the linker search path, because headers
+  # are part of libsystem. Adding its headers to the search path causes strange
+  # mixing and errors.
+  # TODO: Find a better way to prevent this conflict.
+  ncursesInput = if stdenv.isDarwin then ncurses.out else ncurses;
+in
+stdenv.mkDerivation rec {
+  pname = "swift-driver";
+
+  # Releases are made as part of the Swift toolchain, so versions should match.
+  version = "5.7";
+  src = fetchFromGitHub {
+    owner = "apple";
+    repo = "swift-driver";
+    rev = "swift-${version}-RELEASE";
+    hash = "sha256-sk7XWXYR1MGPEeVxA6eA/vxhN6Gq16iD1RHpVstL3zE=";
+  };
+
+  nativeBuildInputs = [ swift swiftpm ];
+  buildInputs = [
+    Foundation
+    XCTest
+    sqlite
+    ncursesInput
+  ];
+
+  patches = [
+    ./patches/nix-resource-root.patch
+    ./patches/disable-catalyst.patch
+    ./patches/linux-fix-linking.patch
+    (fetchpatch {
+      url = "https://patch-diff.githubusercontent.com/raw/apple/swift-driver/pull/1197.patch";
+      hash = "sha256-eVBaKN6uzj48ZnHtwGV0k5ChKjak1tDCyE+wTdyGq2c=";
```

Comment:
This patch has a different SHA now:
> `specified: sha256-eVBaKN6uzj48ZnHtwGV0k5ChKjak1tDCyE+wTdyGq2c=`
> `got:          sha256-NeYu7Ex2F5AgclSP5Y7z507ZxdVMC9XPUlkY64nJIG4=`
Are were sure that this asset is stable?

Reply:
Ah, darn, this is something I'm trying to upstream: https://github.com/apple/swift-driver/pull/1197

I suppose it's not stable until it is merged. I can copy it to nixpkgs, or we can wait a bit? (I'm hoping they'll merge it soon.)

Reply:
Oh that's great. Your call. I'm just trying to do a shakedown on Linux. I don't have the compute resources for the full closure of this PR, but as one of the Linux maintainers thought I should make sure that side was covered. Thanks for this huge contribution!

Reply:
Patch can be stable if referring to individual commit without the /pull/ part of the URL. For example:

https://github.com/apple/swift-driver/commit/d3ef9cdf4871a58eddec7ff0e28fe611130da3f9.patch

It is better when available to reference the merge commit so that it’ll be associated with a branch, but that commit will always be there even if the PR changes.

Reply:
I rebased and used the commit URL. I left a TODO comment; but hope it's otherwise not a blocker?